invoice/src/main/java/org/killbill/billing/invoice/InvoiceDispatcher.java

```diff
@@ -700,4 +705,41 @@ public BillingActionPolicy getBillingActionPolicy() {
             return null;
         }
     }
+
+    public void processParentInvoiceForInvoiceGeneration(final ImmutableAccountData account, final UUID invoiceId, final InternalCallContext context) throws InvoiceApiException {
+
+        final InvoiceModelDao invoiceModelDao = invoiceDao.getById(invoiceId, context);
+        final Invoice invoice = new DefaultInvoice(invoiceModelDao);
+
+        BigDecimal invoiceAmount = InvoiceCalculatorUtils.computeInvoiceBalance(invoice.getCurrency(), invoice.getInvoiceItems(), invoice.getPayments());
+        InvoiceModelDao parentInvoice = invoiceDao.getParentDraftInvoice(account.getParentAccountId(), context);
+
+        final Long parentAccountRecordId = internalCallContextFactory.getRecordIdFromObject(account.getParentAccountId(), ObjectType.ACCOUNT, buildTenantContext(context));
+        final InternalCallContext parentContext = new InternalCallContext(context, parentAccountRecordId);
+
+        final DateTime today = clock.getNow(account.getTimeZone());
+        if (parentInvoice != null) {
+            InvoiceItem invoiceItem = new ParentInvoiceItem(UUID.randomUUID(), today, parentInvoice.getId(), account.getParentAccountId(), account.getId(), invoiceAmount, account.getCurrency());
```

Comment:
An item could already exist at this point, it should be updated instead (each parent invoice should have one item per child per day at most). A good test for it would be to trigger two subscription changes in the same day for a child account.